### DISCUSSION TITLE
New to writing stories in Storbyook for Sveltekit application; What fetch mock library to use?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [storybookjs/storybook#25277] [Help] [answered]
### Summary

I'm new to writing stories, using Sveltekit.

Based on [this guide](https://storybook.js.org/addons/storybook-addon-fetch-mock), what fetch mocking library would you recommend for Sveltekit?

![CleanShot 2023-12-19 at 19 36 44@2x](https://github.com/storybookjs/storybook/assets/9967422/9156ea32-5439-4dd7-9103-b7773f2899b3)

I think the best fit based on the article would be `Mock Service Worker addon` - but is the service worker addon supported with the Sveltekit framework?

**Comment by kettanaito** [comment, marked as answer]
Hi, @alexbjorlig. You can use the [Mock Service Worker add-on for Storybook](https://github.com/mswjs/msw-storybook-addon) to mock network requests in SvelteKit! 

Add-on itself doesn't have to support any particular frameworks. It supports them consequentially since MSW, which the add-on wraps, is framework-agnostic by design. You can use it with Svelte, SvelteKit, Vue, Angular, React, and so on. MSW doesn't ship anything framework-related and, thus, concerns itself not with what framework you are using to develop your application.
